core/domain/question_services.py

```diff
@@ -85,15 +85,16 @@ def _create_new_question(committer_id, question, commit_message):
     create_question_summary(question.id, committer_id)
 
 
-def create_new_question_skill_link(question_id, skill_id):
+def create_new_question_skill_link(question_id, skill_id, skill_difficulty=0.0):
```

Comment:
When creating a new question, are we allowing the learner to set a difficulty or is the difficulty dynamically calculated based on learner responses?
If it is the latter, wouldn't a skill_difficulty of 0, when the question is created, make it seem as if the question is very easy to the first couple of learners, or are we taking 0 as a 'Not Determined' case?

Reply:
When creating a question we will allow the creator to set question difficulty between Easy, Medium, and Hard. Easy will be 0.3 in the backend, Medium = 0.6, Hard = 0.9. Now in the future if we want to dynamically calculate we can update these values whenever required. For the frontend here is how we will identify the difficulty (0 - 0.3: Easy, 0.31 - 0.6: Medium, 0.61 + : Hard)
Now for creators to be able to add difficulty , it would require a change in the UI, so as of now what I have done is pass a constant (0.3), which indicates its an easy question by default. In a future PR I will add difficulty to the UI. Hope this makes sense!

Reply:
Yeah, ok, thanks!